execution/record.go

```diff
@@ -75,9 +132,26 @@ func (r *Record) Equal(other *Record) bool {
 			return false
 		}
 	}
+
+	if !r.metadata.eventTime.Equal(other.metadata.eventTime) {
+		return false
+	}
+
+	if r.metadata.undo != other.metadata.undo {
+		return false
+	}
+
 	return true
 }
 
+func (r *Record) IsUndo() bool {
+	return r.metadata.undo
+}
+
+func (r *Record) EventTimeField() octosql.Value {
```

Comment:
Is this used somewhere? Why not using it in line 73

Reply:
Fixed